src/atoms/buttons/button.test.tsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { render, wait } from "@testing-library/react";
+import { Button } from "./button";
+import faker from "faker";
+import { ButtonSizes } from "../constants/button-sizes";
+import { ButtonStyles } from "../constants/button-styles";
+
+describe("Button", () => {
+    it("when default props, renders button children", async () => {
+        // Arrange
+        const expected = faker.random.words();
+
+        // Act
+        const { getByText } = render(<Button>{expected}</Button>);
+
+        // Assert
+        await wait(() => {
+            expect(getByText(expected)).not.toBeNull();
+        });
+    });
+
+    it("when accessibleText provided, renders child accessible span", () => {
```

Comment:
Nitpick - since these props are inputs passed to the component (not a precondition of the test or component setup), would `given accessibleText ...` make more sense?